webdriver/newsession/create-1.py

```diff
@@ -0,0 +1,24 @@
+#META: timeout=long
+
+import pytest
+from webdriver import error
+
+
+from util.newsession import new_session
+from conftest import product, flatten, platform_name
+
+
+# Note that we can only test things here all implementations must support
+valid_data = [
+    ("proxy", [None]),
+    ("unhandledPromptBehavior", ["dismiss", "accept", None]),
+    ("test:extension", [True, "abc", 123, [], {"key": "value"}, None]),
+]
+
+
+@pytest.mark.parametrize("body", [lambda key, value: {"alwaysMatch": {key: value}},
+                                  lambda key, value: {"firstMatch": [{key: value}]}])
+@pytest.mark.parametrize("key,value", flatten(product(*item) for item in valid_data))
+def test_valid(_function_session, body, key, value):
+    resp = new_session(_function_session, {"capabilities": body(key, value)})
```

Comment:
Writing out the tests would be very repetitive and make it hard to see what's tested. I think we need to use some parameterisation although maybe more explaination is required.